src/main/java/com/management/HumanResources/model/Employee.java

```diff
@@ -19,8 +22,20 @@
     private long managerID;
     private String dept;
 
+    @JsonIgnore
     public String getUniqueData() {
         return (new String(firstName+lastName+email)).toLowerCase();
     }
 
+    public void merge(Employee updatedEmployee) {
+        this.firstName = updatedEmployee.getFirstName() == null ? this.firstName : updatedEmployee.getFirstName();
+        this.lastName = updatedEmployee.getLastName() == null ? this.lastName : updatedEmployee.getLastName();
+        this.email = updatedEmployee.getEmail() == null ? this.email : updatedEmployee.getEmail();
+        this.salary = updatedEmployee.getSalary() == 0 ? this.salary : updatedEmployee.getSalary();
```

Comment:
Could you change the "null" value from 0 to -1 for `salary` and `managerId`? Otherwise you would not be able to remove the manager or remove the salary. e.g. someone becomes the CEO who has no manager.

Reply:
I get where you're coming from. I feel we can treat `0` as a null value in this scenario since that's how Java treats any unassigned primitive number. I think you might be suggesting a breaking change, here's why:
```json
{
    "id": 3479238982331,
    "lastName": "Johnson"
}
```
If the client sent that json to the server it would initialize the Java object like this: 

```java
    id = 3479238982331;
    firstName = null;
    lastName = "Johnson";
    email = null;
    salary = 0;
    position = null;
    address = null;
    managerID = 0;
    dept = null;
```

So setting it to -1 in the checks would let the default 0 value through and override actual data. Realistically nobody should ever have a 0 salary if they work somewhere. Also In the database those who have executive roles have a managerId of `1000000000`

Reply:
Yeah I haven't thought about that.

The special ID is a good solution for the execs, but you would still need to check upon ID creation that it is not equal to that value.

Reply:
Thanks, have updated so that it checks for this